content/docs/pipelines/sdk/build-component.md

```diff
@@ -60,57 +60,65 @@ local file, such as `/output.txt`. In the Python class that defines your
+def dataproc_train_op(
+    project,
+    region,
+    cluster_name,
+    train_data,
+    eval_data,
+    target,
+    analysis,
+    workers,
+    rounds,
+    output,
+    is_classification=True
+):
     if is_classification:
       config='gs://ml-pipeline-playground/trainconfcla.json'
     else:
       config='gs://ml-pipeline-playground/trainconfreg.json'
 
-    super(TrainerOp, self).__init__(
-      name=name,
-      image='gcr.io/ml-pipeline/ml-pipeline-dataproc-train:7775692adf28d6f79098e76e839986c9ee55dd61',
-      arguments=[
-          '--project', project,
-          '--region', region,
-          '--cluster', cluster_name,
-          '--train', train_data,
-          '--eval', eval_data,
-          '--analysis', analysis,
-          '--target', target,
-          '--package', 'gs://ml-pipeline-playground/xgboost4j-example-0.8-SNAPSHOT-jar-with-dependencies.jar',
-          '--workers', workers,
-          '--rounds', rounds,
-          '--conf', config,
-          '--output', output,
-      ],
-      file_outputs={'output': '/output.txt'})
+    return dsl.ContainerOp(
+        name='Dataproc - Train XGBoost model',
+        image='gcr.io/ml-pipeline/ml-pipeline-dataproc-train:ac833a084b32324b56ca56e9109e05cde02816a4',
+        arguments=[
+            '--project', project,
+            '--region', region,
+            '--cluster', cluster_name,
+            '--train', train_data,
+            '--eval', eval_data,
+            '--analysis', analysis,
+            '--target', target,
+            '--package', 'gs://ml-pipeline-playground/xgboost4j-example-0.8-SNAPSHOT-jar-with-dependencies.jar',
+            '--workers', workers,
+            '--rounds', rounds,
+            '--conf', config,
+            '--output', output,
+        ],
+        file_outputs={
+            'output': '/output.txt',
+        }
+    )
 
 ```
 
-The above class is an extract from the
+The function must returns a dsl.ContainerOp from the
 [XGBoost Spark pipeline sample](https://github.com/kubeflow/pipelines/blob/master/samples/xgboost-spark/xgboost-training-cm.py).
 
 Note:
 
 * Each component must inherit from 
   [`dsl.ContainerOp`](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_container_op.py).
-* In the `init` arguments, you can include Python native types (such as `str` 
-  and `int`) and
-  [`dsl.PipelineParam`](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_pipeline_param.py) 
-  types. Each `dsl.PipelineParam` represents a parameter whose value is usually 
-  only known at run time. The parameter can be a one for which the user provides 
-  a value at pipeline run time, or it can be an output from an upstream 
-  component. 
+* Allowed arguments for `dataproc_train_op` include both Python scalar types (such as `str` and ` int`) and           [`dsl.PipelineParam`](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_pipeline_param.py) types when constructing 
```

Comment:
Not quite. The function can take any argument it wants. The allowed arguments restriction only applies to the parameter list passed to the `arguments` argument in dsl.ContainerOp constructor.

Reply:
I think this clicked for me when I finally saw the `arguments` argument in line 91, thank you. Does this read as accurate to you?

"Values passed to the parameter list in the `dsl.ContainerOp`  constructor (project, region, etc.) must be either Python scalar types (such as `str` and ` int`) or `dsl.PipelineParam` types."

Reply:
Yes, that's right. But maybe specify "Values in the `arguments` list that's used by the `dsl.ContainerOp` constructor above must be either ....`

That is, use `arguments` rather than parameter list. Thanks!

Reply:
Done, thank you!